pkg/util/deployment/deployment.go

```diff
@@ -51,32 +50,85 @@ const (
 // GetOldReplicaSets returns the old replica sets targeted by the given Deployment; get PodList and ReplicaSetList from client interface.
 // Note that the first set of old replica sets doesn't include the ones with no pods, and the second set of old replica sets include all old replica sets.
 func GetOldReplicaSets(deployment *extensions.Deployment, c clientset.Interface) ([]*extensions.ReplicaSet, []*extensions.ReplicaSet, error) {
-	return GetOldReplicaSetsFromLists(deployment, c,
+	rsList, err := ListReplicaSets(deployment,
+		func(namespace string, options api.ListOptions) ([]extensions.ReplicaSet, error) {
+			rsList, err := c.Extensions().ReplicaSets(namespace).List(options)
+			return rsList.Items, err
+		})
+	if err != nil {
+		return nil, nil, fmt.Errorf("error listing ReplicaSets: %v", err)
+	}
+	podList, err := ListPods(deployment,
```

Comment:
This is used to separate old RSes that have pods in pod list and those who don't. We use pods list originally because we get RSes from pods that matches deployment's selector. We can use rs's status.replicas to do the same thing. But this can be a follow up PR.